### PR TITLE
Introduce Cypress E2E Testing

### PR DESCRIPTION
**Pull request by idea404**
This PR introduces Cypress E2E Testing

### CODE REVIEW

**Comment by idea404** [comment]
TODO: implement yaml spec

File: .gitignore [hunk 1]
```diff
@@ -9,12 +9,15 @@ node_modules
...
 **/dist
 **/node_modules
+**/.parcel-cache
 
 # keys
 **/neardev
 
 # testing
 /coverage
+**/videos
```
Comment (maxhr) [inline .gitignore:19]: do you need **/videos?

File: frontend/cypress.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const { defineConfig } = require('cypress')
```
Comment (maxhr) [inline frontend/cypress.config.js:1]: can we place this in cypres folder? to keep root folder clean

File: frontend/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (maxhr) [inline frontend/tsconfig.json:1]: why tsconfig in frontend?

File: frontend/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "extends": "../tsconfig.json",
```
Comment (maxhr) [inline frontend/tsconfig.json:2]: this doesn't exist

**Review by maxhr — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (idea404) [reply .gitignore:19]: Cypress makes video backups of runs into this folder

File: frontend/cypress/e2e/guest-book.cy.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+// Load string from environment variable
```
[the remark this answers is not in the document]
Comment (idea404) [reply frontend/cypress/e2e/guest-book.cy.ts:1]: I'd argue snapshot tests belong in the unit tests for the front end. We can add these in another PR that includes tests that test the frontend alone. These tests perform network calls additional to how the frontend works

**Review by maxhr — APPROVED** [review]

**Pull request merged**